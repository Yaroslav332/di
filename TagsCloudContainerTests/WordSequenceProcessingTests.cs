using System.Collections.Generic;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using TagsCloudContainer;

namespace TagsCloudContainerTests;

[TestFixture]
[UseReporter(typeof(DiffReporter))]
public class WordSequenceProcessingTests
{
    [SetUp]
    public void Setup()
    {
    }

    [Test]
    public void WordDistribution_Should_Be_Correct()
    {
        var sequence = new List<string>(
            "AAa aAa B b AA a B b D"
                .Split());

        var wordsHandler = new DefaultWordsHandler(sequence);
        Approvals.VerifyAll(wordsHandler.WordDistribution);
    }


    [Test]
    public void WordDistribution_Should_Be_Correct_WhenEmptyString()
    {
        var sequence = new List<string>();

        var wordsHandler = new DefaultWordsHandler(sequence);
        Approvals.VerifyAll(wordsHandler.WordDistribution);
    }
}