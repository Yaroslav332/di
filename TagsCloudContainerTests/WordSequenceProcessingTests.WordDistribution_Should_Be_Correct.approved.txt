a => 1
aa => 1
aaa => 2
b => 4
d => 1
